java/arcs/android/common/resurrection/ResurrectionRequest.kt

```diff
@@ -37,7 +37,12 @@ data class ResurrectionRequest(
      * If empty, the client requests resurrection for *any* change to *any* [StorageKey]-identified
      * data.
      */
-    val notifyOn: List<StorageKey> = emptyList()
+    val notifyOn: List<StorageKey> = emptyList(),
+    /**
+     * Used to associate [notifierId] with a set of [StorageKeys] and is sent back to listeners
+     * on resurrection along with the keys.
```

Comment:
This doesn't really explain the purpose of the `notifierId` to me as a reader of the code.  NotifierId is intended to allow a resurrection target to create a resurrection request which applies only to changes to the `notifyOn` list coming from _only_ the notifier with the given `notifierId`, right?

Reply:
"coming from only the notifier" The changes can come from anywhere (by that I mean, who is writing to the storage keys can be anything), the notifierId accomplishes two things:

1) multiple clients requesting resurrection with the same component name overwrite each other, This could be two ArcHosts living in the same Activity/Service. ArcHosts themselves need to monitor lots of keys, and then know which arcs to revive.

2) The need for a Service, when woken up, to be given more than just a list of keys, but to be given an associated id (in archost's case, the arcid). This is so that the ArcHost doesn't have to store a map from StorageKey -> ArcId somewhere.  The IntentExtras field was insufficient for this, because it is per-ComponentName, and so every time an ArcHost did requestResurrection() for a new set of keys, it would overwrite the blob from the previous registration.

Reply:
In this case, I feel like the name `notifierId` is a misnomer and is misleading.. It's more of a `targetId` or something (i'm open to alternatives)